pyat/at/plot/resonances.py

```diff
@@ -0,0 +1,505 @@
+    bfarey = 1
+    cfarey = 1
+    dfarey = nthorder
+    farey = [0, 1 / dfarey]
+    fracfarey = [Fraction(0), Fraction(1, dfarey)]
+    idx = 0
+    while (farey[-1] < 1) and (idx < 100):
+        idx += 1
+        caux = numpy.floor((nthorder + bfarey) / dfarey) * cfarey - afarey
+        daux = numpy.floor((nthorder + bfarey) / dfarey) * dfarey - bfarey
+        afarey = cfarey
+        bfarey = dfarey
+        cfarey = int(caux)
+        dfarey = int(daux)
+        farey.append(cfarey / dfarey)
+        fracfarey.append(Fraction(cfarey, dfarey))
+    verboseprint(f"farey_float{nthorder}= {farey}")
+    verboseprint(f"farey_frac{nthorder} = {fracfarey}")
+    return farey, fracfarey
+
+
+def plot_tune2d_resonances(
+    **kwargs: dict[str, any],
+) -> Figure:
+    r"""
+    Plot the tune 2D resonances for a given order, period and window.
+
+    Parameters:
+        kwargs:
+            * orders: integer or list of integers larger than zero. Default: [1,2,3]
+            * period: integer larger than zero; periodicity of the machine. Default: 1.
+            * window: [min_nux,max_nux,min_nuy,max_nuy] list of 4 values for the
+                tune minimum and maximum window. Default:[0,1,0,1].
+            * verbose: print verbose output.
+            * includelegend: print legend on the plot. Default: False.
+            * block: passed to plot.show(). Default: False.
+            * debugverbose: extra output to check line construction. Default: False.
+            * onlyns: if 'n' plots only normal resonances.
+                      if 's' plots only skew resonances.
+                      Otherwise ignored.
+            * custom_linesty: use it to pass a dictionary with custom line styles.
+                See notes below.
+
+    Returns:
+        Figure object
+
+    NOTES:
+    The resonance equation is :math:`a\nu_x + b\nu_y = c`
+    with :math:`a,b` and :math:`c` integers. The order :math:`N=abs(a)+abs(b)`.
+
+    Normal and Skew convention:
+    Line style is similar to reson.m from Matlab Middle Layer, MML, by L. Nadolski.
+    Normal resonances are plotted with a continuous line.
+    Skew resonances, i.e. N-abs(a) is odd, are plotted in dashed lines.
+
+    Color and thickness:
+    Line style is similar to reson.m from Matlab Middle Layer, MML, by L. Nadolski.
+    1st: black, width 4
+    2nd: blue, width 3
+    3rd: red, width 2
+    4th: green, width 1
+    5th: magenta, witdh 1
+    6th: cyan, width 1
+    7th: yellow, width 1
+    8th: darkcyan, width 1
+    9th: 'lightgreen', width 1
+    10th to 15th: RGB increased in steps of 0.1, width 1
+
+    Custom Style:
+    You could pass a custom line style in a dictionary as
+    custom_linesty = mydictionary,
+    where mydictionary should contain two entries
+    dict(0: prop_normal, 1: prop_skew).
+    prop_normal and prop_skew are also dictionaries starting at zero,
+    i.e. the_nth_order-1. Each entry contains the line properties of the nth-1
+    resonance to plot.
```

Comment:
Why not simply `linestyles` for the keyword? (by the way, in the code, you use `customlinesty` without underscore)

Then you could use "normal" and "skew" for the keys of the top-level directory, and "order" instead of "order-1" for the keys of the subdirectories;

Reply:
Done.